environment/src/main/java/com/sequenceiq/environment/parameters/validation/validators/parameter/AzureParameterValidator.java

```diff
@@ -124,11 +127,21 @@ private ValidationResult validateEntitlement(ValidationResultBuilder validationR
                                         + "but that feature is currently disabled").build();
                     }
                 default:
-                    return validationResultBuilder.build();
+                    break;
+            }
+        }
+
+        String keyUrl = azureResourceEncryptionParametersDto.getKeyUrl();
+        if (Objects.nonNull(keyUrl)) {
+            if (!entitlementService.isAzureDiskSSEWithCMKEnabled(accountId)) {
+                LOGGER.info("Invalid request, isAzureDiskSSEWithCMKEnabled entitlement turned off for account {}", accountId);
```

Comment:
I think we should highlight here the entitlement name

Reply:
btw do we need any additional Azure policy for using this feature?

Reply:
I think we should also validate in the current creation flow that the key is really exist on azure side

Reply:
Updated the error to include the Entitlement, `isAzureDiskSSEWithCMKEnabled`.
I don't think additional policies are required, figuring out how to include the key validation, will update either this PR or following one(for CB-11152 only) to have the validation, that's in the to-do list.